kong/cmd/utils/prefix_handler.lua

```diff
@@ -120,6 +121,36 @@ local function compile_conf(kong_config, conf_template)
   end)
 end
 
+local function write_env_file(path, data)
+  local c = require "lua_system_constants"
+
+  local flags = bit.bor(c.O_CREAT(), c.O_WRONLY())
+  local mode  = bit.bor(c.S_IRUSR(), c.S_IWUSR(), c.S_IRGRP())
```

Comment:
pedantic as well since we are dealing with cold code path for now, but since we are doing a `require` and identical bitwise computation on every call, maybe we could cache those in a lexical `do .. end` block around the function. Not a blocker though of course...

Reply:
So this should be compared as a performance vs. readability condition. I'd argue the current format is more readable (what I was going for); performance is not a concern here at all. Happy to adjust this this if needed.

Reply:
Oh really? I personally find it more readable if the function's body is only executing code based on its input, and if all invariant code is left out. But no, no performance concern here, so I think this is fine as is if you prefer this form :+1:

Reply:
Pedantic comment:

> maybe we could cache those in a lexical `do .. end` block around the function

This is not a cache, but a localization trick for the compiler (transpiler?), to my understanding. To get a bit pedantic, I would say a `do ... end` block is harder to read, mentally, so the current form is acceptable. But, I am 100% on board with writing referential code, e.g., code that can be used for style/perf references in the future, so I'm happy to make adjustments. We're getting quite "weedy" here? A good point for discussions, though :+1: I have no strong feeling either way, given this is a cold path and we are noting non-functional differences.

Reply:
> But, I am 100% on board with writing referential code

Yep, this is very much my goal most of the time when pointing those out.